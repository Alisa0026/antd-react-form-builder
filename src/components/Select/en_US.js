export default {
  FormBuilderSelect: {
    placeholder: "Please input",
    yes: "yes",
    no: "no",
  },
}

