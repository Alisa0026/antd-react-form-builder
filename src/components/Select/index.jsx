import React from 'react'
import BasicItem from '../BasicItem'
import AntdSelect from 'antd/lib/select'

const Option = AntdSelect.Option;
const OptGroup = AntdSelect.OptGroup;

function component(BasicItemComponent){
  return class Select extends React.Component {
    render(){
      let {
        options,
        group,
        ...other, 
      } = this.props;
      other.targetComponent = AntdSelect;
      if(group){
        return (
          //这里的BasicItemComponent相当于
          //<FormItem>
          //  <Select>(AntdSelect)
          //    { children }
          //  </Select>
          //</FormItem>
          <BasicItemComponent { ...other }>
            {
              group && group[0] && group.map((g_v,g_k)=>{
                return (
                  <OptGroup key={ g_k } label={ g_v.label || "" }>
                    {
                      g_v.options && g_v.options[0] && g_v.options.map((o_v,o_k)=>{
                        return (
                          <Option key={ o_k } value={ o_k+"" }>
                            { o_v.text }
                          </Option>
                        )
                      })
                    }
                  </OptGroup>
                )
              })
            }
          </BasicItemComponent> 
        )
      }else if(options){
        return (
          <BasicItemComponent { ...other }>
            {
              options && options[0] && options.map((o_v,o_k)=>{
                return (
                  <Option key={ o_k } value={ o_k+"" }>
                    { o_v.text }
                  </Option>
                )
              })
            }
          </BasicItemComponent>
        )
      }
    }
  }
}
export default component(BasicItem) 
