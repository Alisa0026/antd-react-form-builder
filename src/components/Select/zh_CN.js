export default {
  FormBuilderSelect: {
    placeholder: "请输入",
    yes: "是",
    no: "否",
  },
}


