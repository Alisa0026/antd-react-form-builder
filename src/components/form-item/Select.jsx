import React from 'react'
import AntdSelect from 'antd/lib/select'
import AntdForm from 'antd/lib/form'

let FormItem = AntdForm.Item;
let Option = AntdSelect.Option;

export default class Select extends React.Component {

  static contextTypes = {
    form: React.PropTypes.any
  }

  constructor(props){
    super(props);
  }

  getRules(type){
    return [];
  }

  render() {
    let props = this.props;
    let {
      value,
      name,
      rules = [],
      options,
      formItemProps,
      ...other,
    } = props;
    let form = this.context.form;
    if(!form){
      console.error("请使用Antd.Form.create()(targetForm)处理目标form函数对象（类）")
      return false;
    }

    Array.prototype.push.apply(rules,this.getRules())
    var obj = {
      rules: rules || [],
    }
    if(value || value == 0){
      obj.initialValue = value; 
    }
    var element = form.getFieldDecorator(name, obj)(
      <AntdSelect {...other}>
        {
          options && options[0] && options.map((o_v,o_k)=>{
            return (
              <Option key={ o_k } value={ o_k+"" }>
                { o_v.text }
              </Option>
            )
          })
        }
      </AntdSelect>
    )

    return (
      <FormItem {...formItemProps}>
        { element }
      </FormItem>
    )
  }
}


