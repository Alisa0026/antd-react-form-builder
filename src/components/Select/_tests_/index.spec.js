import React from 'react'
import { shallow,mount,render } from 'enzyme'
import sinon from 'sinon'
import LocaleProvider from "antd/lib/locale-provider"
import { 
  testCommonProps, 
  getAntdFormComponentMount,
  getAntdFormItemMount,
} from '../../../../tests/common'
import Select from '../index'

var options = [
  {
    value: "shenzhen",
    label: "深圳"
  },
  {
    value: "beijing",
    label: "北京"
  },
  {
    value: "hangzhou",
    label: "hangzhou"
  },
];
var group = [
  {
    label: "广东省",
    options: [
      {
        value: "shenzen",
        label: "深圳"
      }
    ] 
  }
];

describe('Select',function(){
  sinon.stub(console,"warn");
  it('1 renders correctly', () => {
    const wrapper = render(
      <Select 
        required
        className="test"
        label="name"
        name="test"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('2 renders correctly', () => {
    const wrapper = render(
      <Select 
        required
        className="test"
        label="name"
        name="test"
        boolean={ true }
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('3 renders correctly', () => {
    const wrapper = render(
      <Select 
        required
        className="test"
        label="name"
        name="test"
        options={ options }
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('4 renders correctly', () => {
    const wrapper = render(
      <Select 
        required
        className="test"
        label="name"
        name="test"
        options={ options }
        mutiple={true}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('5 renders correctly', () => {
    const wrapper = render(
      <Select 
        required
        className="test"
        label="name"
        name="test"
        group={ group }
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  testCommonProps(Select); 
  it("props.boolean=true should have two <Option />",function(){
    const wrapper = mount(
      <Select 
        required
        name="test"
        label="label"
        boolean={ true }
      />
    );
    const antdInputMount = getAntdFormComponentMount(wrapper,1);
    const antdFormItemMount = getAntdFormItemMount(wrapper);
    expect(antdInputMount.props().children.length).toEqual(2);
  })
  it("should have three <Option />",function(){
    const wrapper = mount(
      <Select 
        required
        name="test"
        label="label"
        options={ options }
      />
    );
    const antdInputMount = getAntdFormComponentMount(wrapper,1);
    const antdFormItemMount = getAntdFormItemMount(wrapper);
    expect(antdInputMount.props().children.length).toEqual(3);
  })
  it("should have one <Group />",function(){
    const wrapper = mount(
      <Select 
        required
        name="test"
        label="label"
        group={ group }
      />
    );
    const antdInputMount = getAntdFormComponentMount(wrapper,1);
    const antdFormItemMount = getAntdFormItemMount(wrapper);
    expect(antdInputMount.props().children.length).toEqual(1);
  })
})

