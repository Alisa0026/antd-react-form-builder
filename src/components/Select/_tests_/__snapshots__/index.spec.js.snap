// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select 1 renders correctly 1`] = `
<span>
  <div
    class="ant-row ant-form-item"
  >
    <div
      class="ant-form-item-label"
    >
      <label
        class="ant-form-item-required"
        title="name"
      >
        name
      </label>
    </div>
    <div
      class="ant-form-item-control-wrapper"
    >
      <div
        class="ant-form-item-control "
      >
        <div
          class="test ant-select ant-select-enabled"
        >
          <div
            aria-autocomplete="list"
            aria-expanded="false"
            aria-haspopup="true"
            class="ant-select-selection
                      ant-select-selection--single"
            role="combobox"
            tabindex="0"
          >
            <div
              class="ant-select-selection__rendered"
            >
              <div
                class="ant-select-selection__placeholder"
                style="display:block;user-select:none;-webkit-user-select:none;"
                unselectable="unselectable"
              >
                请输入
              </div>
            </div>
            <span
              class="ant-select-arrow"
              style="user-select:none;-webkit-user-select:none;"
              unselectable="unselectable"
            >
              <b />
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
  <input
    class="ant-input"
    name="test"
    type="hidden"
    value=""
  />
</span>
`;

exports[`Select 2 renders correctly 1`] = `
<span>
  <div
    class="ant-row ant-form-item"
  >
    <div
      class="ant-form-item-label"
    >
      <label
        class="ant-form-item-required"
        title="name"
      >
        name
      </label>
    </div>
    <div
      class="ant-form-item-control-wrapper"
    >
      <div
        class="ant-form-item-control "
      >
        <div
          class="test ant-select ant-select-enabled"
        >
          <div
            aria-autocomplete="list"
            aria-expanded="false"
            aria-haspopup="true"
            class="ant-select-selection
                      ant-select-selection--single"
            role="combobox"
            tabindex="0"
          >
            <div
              class="ant-select-selection__rendered"
            >
              <div
                class="ant-select-selection__placeholder"
                style="display:block;user-select:none;-webkit-user-select:none;"
                unselectable="unselectable"
              >
                请输入
              </div>
            </div>
            <span
              class="ant-select-arrow"
              style="user-select:none;-webkit-user-select:none;"
              unselectable="unselectable"
            >
              <b />
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
  <input
    class="ant-input"
    name="test"
    type="hidden"
    value=""
  />
</span>
`;

exports[`Select 3 renders correctly 1`] = `
<span>
  <div
    class="ant-row ant-form-item"
  >
    <div
      class="ant-form-item-label"
    >
      <label
        class="ant-form-item-required"
        title="name"
      >
        name
      </label>
    </div>
    <div
      class="ant-form-item-control-wrapper"
    >
      <div
        class="ant-form-item-control "
      >
        <div
          class="test ant-select ant-select-enabled"
        >
          <div
            aria-autocomplete="list"
            aria-expanded="false"
            aria-haspopup="true"
            class="ant-select-selection
                      ant-select-selection--single"
            role="combobox"
            tabindex="0"
          >
            <div
              class="ant-select-selection__rendered"
            >
              <div
                class="ant-select-selection__placeholder"
                style="display:block;user-select:none;-webkit-user-select:none;"
                unselectable="unselectable"
              >
                请输入
              </div>
            </div>
            <span
              class="ant-select-arrow"
              style="user-select:none;-webkit-user-select:none;"
              unselectable="unselectable"
            >
              <b />
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
  <input
    class="ant-input"
    name="test"
    type="hidden"
    value=""
  />
</span>
`;

exports[`Select 4 renders correctly 1`] = `
<span>
  <div
    class="ant-row ant-form-item"
  >
    <div
      class="ant-form-item-label"
    >
      <label
        class="ant-form-item-required"
        title="name"
      >
        name
      </label>
    </div>
    <div
      class="ant-form-item-control-wrapper"
    >
      <div
        class="ant-form-item-control "
      >
        <div
          class="test ant-select ant-select-enabled"
        >
          <div
            aria-autocomplete="list"
            aria-expanded="false"
            aria-haspopup="true"
            class="ant-select-selection
                      ant-select-selection--single"
            role="combobox"
            tabindex="0"
          >
            <div
              class="ant-select-selection__rendered"
            >
              <div
                class="ant-select-selection__placeholder"
                style="display:block;user-select:none;-webkit-user-select:none;"
                unselectable="unselectable"
              >
                请输入
              </div>
            </div>
            <span
              class="ant-select-arrow"
              style="user-select:none;-webkit-user-select:none;"
              unselectable="unselectable"
            >
              <b />
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
  <input
    class="ant-input"
    name="test"
    type="hidden"
    value=""
  />
</span>
`;

exports[`Select 5 renders correctly 1`] = `
<span>
  <div
    class="ant-row ant-form-item"
  >
    <div
      class="ant-form-item-label"
    >
      <label
        class="ant-form-item-required"
        title="name"
      >
        name
      </label>
    </div>
    <div
      class="ant-form-item-control-wrapper"
    >
      <div
        class="ant-form-item-control "
      >
        <div
          class="test ant-select ant-select-enabled"
        >
          <div
            aria-autocomplete="list"
            aria-expanded="false"
            aria-haspopup="true"
            class="ant-select-selection
                      ant-select-selection--single"
            role="combobox"
            tabindex="0"
          >
            <div
              class="ant-select-selection__rendered"
            >
              <div
                class="ant-select-selection__placeholder"
                style="display:block;user-select:none;-webkit-user-select:none;"
                unselectable="unselectable"
              >
                请输入
              </div>
            </div>
            <span
              class="ant-select-arrow"
              style="user-select:none;-webkit-user-select:none;"
              unselectable="unselectable"
            >
              <b />
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
  <input
    class="ant-input"
    name="test"
    type="hidden"
    value=""
  />
</span>
`;
